iceoryx_examples/waitset/README.md

```diff
@@ -1,6 +1,566 @@
+|wait for triggers           |`auto triggerVector = myWaitSet.wait();`  |
+|wait for triggers with timeout |`auto triggerVector = myWaitSet.timedWait(1_s);`  |
+|check if trigger originated from some class|`trigger.doesOriginateFrom(ptrToSomeClass)`|
+|get id of trigger|`trigger.getTriggerId()`|
+|call triggerCallback|`trigger()`|
+|acquire _TriggerOrigin|`trigger.getOrigin<OriginType>();`|
+|check if 2 trigger are logical equal|`trigger.isLogicalEqualTo(anotherTrigger)`|
+
+## Use cases
+This example consists of 5 use cases.
+ 
+ 1. `ice_waitset_gateway.cpp`: We build a gateway together to forward data
+    to another network. A list of subscribers is handled in an uniform way 
+    by defining a callback and just call it for every subscriber who received data.
+
+ 2. `ice_waitset_grouping`: We would like to group multiple subscribers into 2 
+    groups and handle them according to their group membership.
+
+ 3. `ice_waitset_individual`: A list of subscribers where every subscriber is 
+    handled differently.
+
+ 4. `ice_waitset_sync`: We use the WaitSet to trigger a cyclic call which should 
+    execute an algorithm every 100ms.
+
+ 5. `ice_waitset_trigger`: We create our own class which can be attached to a
+    WaitSet and trigger events.
+
+## Examples
+
+All our examples require a running `iox-roudi` and a running `iox-ex-waitset-publisher`
+which is sending the data. The publisher does not contain any _WaitSet_ specific
+logic and is explained in detail in the [icedelivery example](../icedelivery/).
+
+### Gateway
+
+We have a list of subscribers which can be subscribed to any arbitrary service
+and everytime we received a sample we would like to send the bytestream to a socket,
+write it into a file or print it to the console. But whatever we choose to do
+we perform the same task for all the subscribers.
+
+Lets start by implementing our callback which prints the subscriber pointer, the
+payload size and the payload pointer to the console.
+```cpp
+void subscriberCallback(iox::popo::UntypedSubscriber* const subscriber)
+{
+    subscriber->take().and_then([&](iox::popo::Sample<const void>& sample) {
+        std::cout << "subscriber: " << std::hex << subscriber << " length: " << std::dec
+                  << sample.getHeader()->m_info.m_payloadSize << " ptr: " << std::hex << sample.getHeader()->payload()
+                  << std::endl;
+    });
+}
+```
+A _Trigger_ always requires a callback which has the following signature 
+`void (TriggerOrigin)`. In our example the _TriggerOrigin_ is a
+`iox::popo::UntypedSubscriber*` which we use to acquire the latest sample which the subscriber
+has received (`take()`). When `take()` was successful we print our message to
+the console inside of the `and_then` lambda.
+
+In our `main` function we create a _WaitSet_ after we registered at our central
+broker RouDi and attach our `shutdownGuard` to it to handle `CTRL+c` events. 
```

Comment:
What about `shutdownTrigger`? Would seem more consistent to me, if we decided to stick with the next gen terminology.

Reply:
Yes you are right. Some remainders from the past ...